moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,51 @@ bool World::hasObject(const std::string& id) const
   return objects_.find(id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& id) const
+{
+  // Check object names first
+  if (objects_.find(id) != objects_.end())
+    return true;
+  else  // Then objects' named frames
+  {
+    for (auto o : objects_)
+    {
+      if (o.second->named_frames_.count(id) > 0)
+        return true;
+    }
+  }
+  return false;
+}
+
+const Eigen::Affine3d& World::getTransform(const std::string& id) const
+{
+  auto it = objects_.find(id);
+  if (it != objects_.end())
+    return it->second->shape_poses_[0];
+  else  // Find within named frames
+  {
+    for (auto o : objects_)
+    {
+      if (o.second->named_frames_.count(id) > 0)
+        return o.second->named_frames_[id];
+    }
+  }
+}
+
+std::string World::getParent(const std::string& id) const
+{
+  for (auto o : objects_)
+  {
+    if (o.second->named_frames_.count(id) > 0)
+      return o.second->id_;
+  }
+  // If it is an object's name, return the object
+  auto it = objects_.find(id);
```

Comment:
Here, you change the order (now first searching for object frames, then for objects).
In all other functions, you did it the other way around. This can introduce bugs with non-unique names as well.

Reply:
True. It's because this is the only one I wrote from scratch, as a simple getter function. I am not sure it is necessary at all. I changed the order, but I would also be ok with deleting the function.

Reply:
Reordered this as requested.